pkg/virt-handler/vm.go

```diff
@@ -38,8 +40,29 @@ func NewVMController(lw cache.ListerWatcher, domainManager virtwrap.DomainManage
 				queue.AddRateLimited(key)
 				return true
 			}
-
 			vm = v1.NewVMReferenceFromName(name)
+
+			// If we don't have the VM in the cache, it could be that it is currently migrating to us
+			result := restClient.Get().Name(vm.GetObjectMeta().GetName()).Resource("vms").Namespace(kubeapi.NamespaceDefault).Do()
+			if result.Error() == nil {
+				// So the VM still seems to exist
+				fetchedVM, err := result.Get()
+				if err != nil {
+					// Since there was no fetch error, this should have worked, let's back off
+					queue.AddRateLimited(key)
+					return true
+				}
+				if fetchedVM.(*v1.VM).Status.MigrationNodeName == host {
+					// OK, this VM is migrating to us, don't interrupt it
+					queue.Forget(key)
+					return true
+				}
+			} else if result.Error().(*errors.StatusError).Status().Code != int32(http.StatusNotFound) {
+				// Something went wrong, let's try again later
+				queue.AddRateLimited(key)
```

Comment:
To be completely precise we probably have to check if it is a server error 5xx (then always retry), or if it is an 4xx error which is not 404 ... It does probably not make sense to retry a 400 or a 422 ...